### PR TITLE
Add Aggregate Expression support.

### PR DESCRIPTION
**Pull request by telser**
- Added support for aggregates in the Expr level.

- Added helpers for many aggregate functions that were not already supported as window functions

- Changed window functions to reflect some functions are usable in aggregate and window context.

- Added helpers for the aggregate functions to be used as window functions, where applicable.

It's also worth noting that we really should change the `COUNT` support to be part of this world now too. I've left that for the future though.

### CODE REVIEW

File: orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+aggregateStarExpression function mbWhereClause =
+  let
+    parameters = pure . RawSql.unsafeFromRawSql $ RawSql.fromString "*"
+    filtering =
+      case mbWhereClause of
+        Nothing -> mempty
+        Just whereClause ->
+          RawSql.space
+            <> RawSql.fromString "FILTER"
+            <> RawSql.parenthesized (RawSql.toRawSql whereClause)
```
[the remark this answers is not in the document]
Comment (telser) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs:118]: That's a great point.

File: orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        <> RawSql.parenthesized
+          ( case mbAggOption of
+              Nothing -> mempty
+              Just aggOption ->
+                RawSql.toRawSql aggOption
+                  <> RawSql.space
+              <> RawSql.intercalate RawSql.comma parameters
+              <> case mbOrderByClause of
+                Nothing -> mempty
+                Just orderByClause -> RawSql.toRawSql orderByClause
```
[the remark this answers is not in the document]
Comment (ysangkok) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs:89]: Wouldn't fourmolu just undo this suggestion again?  AFAIK fourmolu doesn't enforce that the case branches are indented further than the case.

[the remark this answers is not in the document]
Comment (jlavelle) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs:89]: Oh maybe, I was more concerned with the leading `<>` indentation though.

[the remark this answers is not in the document]
Comment (telser) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs]: The suggested indentation is incorrect though. We don't need to be in the just case to append the parameters or the orderby. Addressed this with some parens to make it more clear.

File: orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (telser) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs]: This should be fixed now

File: orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlavelle) [reply orville-postgresql/src/Orville/PostgreSQL/Expr/Aggregate.hs]: Oh lol, I didn't even notice that.

**Pull request merged**